improver_tests/utilities/test_field_texture.py

```diff
@@ -0,0 +1,171 @@
+        on an equal area grid with 2km grid spacing."""
+
+    cloud_area_fraction = np.zeros((1, 10, 10), dtype=np.float32)
+    cube = set_up_probability_cube(
+        cloud_area_fraction,
+        variable_name="cloud_area_fraction",
+        threshold_units="1",
+        thresholds=[0.8125],
+        spatial_grid="equalarea",
+    )
+    cube = add_coordinate(cube, [0, 1, 2], "realization", dtype=np.int32)
+    cube = iris.util.squeeze(cube)
+    return cube
+
+
+@pytest.fixture(name="all_cloud_cube")
+def all_cloud_fixture():
+    """Manipulate an existing multi-realization cube so the field is occupied
+       entirely by cloud."""
+
+    cloud_data = np.ones((3, 10, 10), dtype=np.float32)
+    cube = set_up_variable_cube(cloud_data, name="cloud_data", spatial_grid="equalarea")
+    return cube
+
+def test__calculate_ratio(multi_cloud_cube):
+    """Test the _calculate_ratio function with one realization of the input cube"""
+
+    expected_data = np.where(multi_cloud_cube.data[0] == 0.0, 1.0, 0.3333)
+
+    plugin = FieldTexture(nbhood_radius=NB_RADIUS, ratio_threshold=RATIO_THRESH)
+    ratio_result = plugin._calculate_ratio(multi_cloud_cube[0], NB_RADIUS)
+    np.testing.assert_almost_equal(ratio_result.data, expected_data, decimal=4)
+
+
+def test_process(multi_cloud_cube):
+    """Test the _calculate_clumpiness function with multi realization input cube"""
+
+    expected_data = np.where(multi_cloud_cube.data[0] == 0.0, 1.0, 0.0)
+
+    plugin = FieldTexture(nbhood_radius=NB_RADIUS, ratio_threshold=RATIO_THRESH)
+    clumpiness_result = plugin.process(multi_cloud_cube)
+    np.testing.assert_almost_equal(clumpiness_result.data, expected_data, decimal=4)
+
+
+def test_process_error(multi_cloud_cube):
+    """Test the _calculate_clumpiness function with a single realization input
+       cube to raise an error."""
+
+    expected_data = np.where(multi_cloud_cube.data[0] == 0.0, 2.0, 0.0)
+
+    plugin = FieldTexture(nbhood_radius=NB_RADIUS, ratio_threshold=RATIO_THRESH)
+    clumpiness_result = plugin.process(multi_cloud_cube)
+    assert (clumpiness_result, "Incorrect input. Cube should hold binary data only")
+
+def test__calculate_transitions(multi_cloud_cube):
+    """Test the _calculate_transitions function with a numpy array simulating
+       the multi-realization input cube."""
+
+    expected_data = np.zeros((10, 10), dtype=np.float32)
+    expected_data[1:4, 1:4] = np.array(
+        [[2.0, 1.0, 2.0], [1.0, 0.0, 1.0], [2.0, 1.0, 2.0]]
+    )
+    plugin = FieldTexture(nbhood_radius=NB_RADIUS, ratio_threshold=RATIO_THRESH)
+    transition_result = plugin._calculate_transitions(multi_cloud_cube[0].data)
+    np.testing.assert_almost_equal(transition_result, expected_data, decimal=4)
+
+
+def test_process_no_cloud(no_cloud_cube):
+    """Test the FieldTexture plugin with multi realization input cube that has
+       no cloud present in the field."""
+
+    expected_data = np.where(no_cloud_cube.data[0] == 0.0, 1.0, 0.0)
```

Comment:
Is this essentially `expected_data = np.zeros((10, 10))`?  Might be clearer to put it that way, if so.

Reply:
This would be the same as writing (in your method): 

expected_data = np.ones(no_cloud_cube[0].shape) 

I'll change it to that because that looks more elegant.